DynamicData/List/ObservableListEx.cs

```diff
@@ -1060,6 +1061,90 @@ public static IObservable<IChangeSet<T>> Or<T>([NotNull] this IObservable<IChang
             return source.Combine(CombineOperator.Or, others);
         }
 
+        /// <summary>
+        /// Performs a dynamic concatentation of the input lists. Order will be preserved but
+        /// the performance may suck a little.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="sources"></param>
+        /// <returns></returns>
+        public static IObservableList<T> DynamicConcatOrderPreserving<T>
+            ([NotNull] this IObservableList<IObservableList<T>> sources)
+        {
+            var foo = sources
+                .Connect()
+                .ToCollection()
+                .Select
+                (collection =>
+                 {
+                     var enumerable = collection
+                         .Select(a => a.Connect().ToCollection())
+                         .ToList();
+                     var observables = enumerable.Count == 0 
+                     ? Observable.Return(new List<T>())
+                     : enumerable
+                         .CombineLatest(list => list.SelectMany(p => p).ToList());
+
+                     return observables;
+
+                 })
+                .Switch()
+                .StartWith(new List<T>())
+                .Buffer(2, 1)
+                .Where(b=>b.Count==2)
+                .Select
+                (pair =>
+                 new ChangeSet<T>
+                     (new[]
+                      {
+                          new Change<T>(ListChangeReason.Clear, pair[0]), new Change<T>(ListChangeReason.AddRange, pair[1])
+                      }));
+
+            return foo.AsObservableList();
+
+        } 
+
+        /// <summary>
+        /// Apply a concatenation of all the lists without preserving order. Child
+        /// lists can be added and removed
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="sources"></param>
+        /// <returns></returns>
+        public static IObservableList<T> DynamicConcatNonOrderPreserving<T>
```

Comment:
I presume you mean concat because of the way it is implemented.  I would suggest the final version will only need one method as order should always be preserved